pkg/operator/ceph/cluster/watcher.go

```diff
@@ -67,8 +79,9 @@ func (c *clientCluster) onK8sNode(object runtime.Object) bool {
 		return false
 	}
 
-	if !cluster.Spec.Storage.UseAllNodes {
-		logger.Debugf("node watcher: do not use all Nodes in cluster %q, skipping", cluster.Namespace)
+	if !checkStorageForNode(cluster) {
+		nodesCheckedForReconcile.Add(node.Name)
+		logger.Debugf("node watcher: useAllNodes is set to false and no nodes storageClassDevicesets or volumeSources are specified in cluster %q, skipping", cluster.Namespace)
```

Comment:
how about adding this inside the `checkStorageForNode()` method?

Reply:
Yes, it will be good to have it there.